### ISSUE TITLE
[apex] ApexCRUDViolationRule fails to report CRUD violation on COUNT() queries

### ISSUE THREAD

**Issue by jonathanwiesel** [pmd/pmd#3202]
**Affects PMD Version:** 
5.5.4 - 6.33

**Rule:**

[ApexCRUDViolation](https://pmd.github.io/latest/pmd_rules_apex_security.html#apexcrudviolation)

**Description:**

A query that uses a simple `COUNT()`with no other aggregators and no fields inside the `COUNT` operations fails to report the CRUD check.

**Code Sample demonstrating the issue:**

```
/**
 * @description X
 */
public with sharing class CRUDCheck {
    
    /**
     * @description X
     */
    public static void checkMe() {

        //this gets reported
        Integer x = [SELECT COUNT(Id) FROM Account];

        //this doesnn't get reported
        Integer y = [SELECT COUNT() FROM Account];
    }
}
```

**Expected outcome:**

*   Is PMD missing to report a violation, where there should be one? -> false-negative

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Apex PMD VSCode Extension

**Comment by jonathanwiesel** [comment]
I see that this is [designed on purpose](https://github.com/pmd/pmd/commit/52651c0c3f9f8e202df9c9cfedeb73d752c12208), however I don't quite understand why.

**Comment by adangel** [comment]
I think you are right. Even `COUNT()` is accessing the data and the rule even states that no checking access permission even "may produce runtime errors"... It also doesn't make sense, why `COUNT()` should be allowed without access permission but `COUNT(column)` not...
